src/wallet/wallet.cpp

```diff
@@ -2993,8 +2986,8 @@ bool CWallet::TopUpKeyPool(unsigned int kpSize)
 
         // count amount of available keys (internal, external)
         // make sure the keypool of external and internal keys fits the user selected target (-keypool)
-        int64_t amountExternal = KeypoolCountExternalKeys();
-        int64_t amountInternal = setKeyPool.size() - amountExternal;
+        int64_t amountExternal = setExternalKeyPool.size();
```

Comment:
nit: `amountExternal` is not longer required, `setExternalKeyPool.size()` can be used directly at L2991+.